base/utf16.jl

```diff
@@ -39,7 +29,7 @@ function next(s::UTF16String, i::Int)
     elseif length(s.data)-1 > i && utf16_is_lead(s.data[i]) && utf16_is_trail(s.data[i+1])
         return utf16_get_supplementary(s.data[i], s.data[i+1]), i+2
     end
-    throw(ArgumentError("invalid UTF-16 character index"))
+    throw(UnicodeError(UTF_ERR_INVALID_INDEX,0,0))
```

Comment:
I've actually got a new idea this morning and would like to try out. (Sth like a `throw` local noinline). Maybe that can bring down the gap between #11284 and #11508 .